### PR TITLE
Audio dumper: append filename with titleid and date-time

### PR DESCRIPTION
**Pull request by illusion0001**

### CODE REVIEW

**Comment by Megamouse** [comment]
You might also add the title id, if it helps.

**Comment by illusion0001** [comment]
added title id before date and time

File: rpcs3/Emu/Audio/AudioDumper.cpp [hunk 1]
```diff
@@ -1,13 +1,15 @@
 ﻿#include "stdafx.h"
 #include "AudioDumper.h"
 
+#include "Utilities/date_time.h"
+#include "Emu/Cell/lv2/sys_event.h"
```
Comment (Megamouse) [inline rpcs3/Emu/Audio/AudioDumper.cpp:5]: what'S this include for?

Comment (illusion0001) [reply rpcs3/Emu/Audio/AudioDumper.cpp:5]: it won't build for some reason with `'Emu': undeclared identifier` so i have to add it

Comment (Megamouse) [reply rpcs3/Emu/Audio/AudioDumper.cpp:5]: but the proper import for Emulator should be
`#include "Emu/System.h"`

Comment (illusion0001) [reply rpcs3/Emu/Audio/AudioDumper.cpp]: thanks, fixed

File: rpcs3/Emu/Audio/AudioDumper.cpp [hunk 2]
```diff
@@ -1,13 +1,15 @@
 ﻿#include "stdafx.h"
 #include "AudioDumper.h"
 
+#include "Utilities/date_time.h"
+#include "Emu/system.h"
```
Comment (Megamouse) [inline rpcs3/Emu/Audio/AudioDumper.cpp:5]:
```suggestion
#include "Emu/System.h"
```

Comment (illusion0001) [reply rpcs3/Emu/Audio/AudioDumper.cpp]: 👍

**Comment by AniLeo** [comment]
Did you test with standalone executables that don't have a Game ID?

**Comment by elad335** [comment]
Emu.GetTitleID() returns a zero sized string in this case iirc, may want to drop one '_' in this case.

**Comment by illusion0001** [comment]
Just tested with elf and yeah it shows `audio__2020...` and normal executable `audio_BCES00509_2020...`if I drop on of the underscore it would look like `audioBCES00509_2020...` and vice versa.

**Comment by elad335** [comment]
I said drop it *when* there is no game id.
it looks like `audio_BCES00509_2020` with game id and `audio_2020` without.

File: rpcs3/Emu/Audio/AudioDumper.cpp [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+#include "Utilities/date_time.h"
+#include "Emu/System.h"
 
 AudioDumper::AudioDumper(u16 ch)
 	: m_header(ch)
 {
 	if (GetCh())
 	{
-		m_output.open(fs::get_cache_dir() + "audio.wav", fs::rewrite);
+		m_output.open(fs::get_cache_dir() + "audio_" + Emu.GetTitleID() + "_" + date_time::current_time_narrow<'_'>() + ".wav", fs::rewrite);
```
Comment (Megamouse) [inline rpcs3/Emu/Audio/AudioDumper.cpp:12]: For example like this:
```suggestion
		std::string path = fs::get_cache_dir() + "audio_";
		if (const std::string id = Emu.GetTitleID(); !id.empty())
		{
			path += id + "_";
		};
		path += date_time::current_time_narrow<'_'>() + ".wav";
		m_output.open(path, fs::rewrite);
```

Comment (illusion0001) [reply rpcs3/Emu/Audio/AudioDumper.cpp]: Thank you

**Comment by AniLeo** [comment]
The latest commit still doesn't seem to avoid the extra underscore

**Comment by illusion0001** [comment]
Oops, rebase issue will fix

File: rpcs3/Emu/Audio/AudioDumper.cpp [hunk 4]
```diff
@@ -1,13 +1,21 @@
...
 AudioDumper::AudioDumper(u16 ch)
 	: m_header(ch)
 {
 	if (GetCh())
 	{
-		m_output.open(fs::get_cache_dir() + "audio.wav", fs::rewrite);
+		std::string path = fs::get_cache_dir() + "audio_";
+		if (const std::string id = Emu.GetTitleID(); !id.empty())
+		{
+			path += id + "_";
```
Comment (AniLeo) [inline rpcs3/Emu/Audio/AudioDumper.cpp:15]: As is this will make path = `audio__`

**Pull request merged**